pageserver/src/tenant/vectored_blob_io.rs

```diff
@@ -102,7 +102,22 @@ impl VectoredReadBuilder {
             start: start_offset,
             end: end_offset,
             blobs_at,
-            max_read_size,
+            max_read_size: Some(max_read_size),
+        }
+    }
+
+    #[cfg(test)]
+    fn new_with_controlled_limit(start_offset: u64, end_offset: u64, meta: BlobMeta) -> Self {
```

Comment:
Why do we need to constructors for? We could just use `VectoredReadBuilder::new(... max_read_size: Option<> ...)`

Reply:
I changed Option to the inner type for the normal VectoredReadBuilder::new constructor

Reply:
Not sure I understand. For clarity, I'm suggesting we remove this constructor.

Reply:
The signature of `new` is

```
pub(crate) fn new(
        start_offset: u64,
        end_offset: u64,
        meta: BlobMeta,
        max_read_size: usize,
    ) -> Self {
```

where it takes `max_read_size` as `usize` to avoid users from misusing that.

and `fn new_with_controlled_limit(start_offset: u64, end_offset: u64, meta: BlobMeta) -> Self {` will store `None` to the `max_read_size` field.

Anyways, I can remove that as the streaming vectored read struct can access private fields of vectored read in the same file.

Reply:
Yeah, i get that. So is there a reason why we don't want the signature to be:
```
pub(crate) fn new(
        start_offset: u64,
        end_offset: u64,
        meta: BlobMeta,
        max_read_size: Option<usize>,
    ) -> Self {
```